joinfaces-autoconfigure/src/main/java/org/joinfaces/autoconfigure/rewrite/SpringBootServiceLocator.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2016-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.joinfaces.autoconfigure.rewrite;
+
+import java.util.Collection;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Set;
+
+import org.ocpsoft.common.spi.ServiceLocator;
+
+import org.springframework.context.ApplicationContext;
+
+/**
+ * {@link ServiceLocator} implementation for Spring. Inspired by
+ * https://github.com/ocpsoft/rewrite/blob/master/integration-spring/src/main/java/org/ocpsoft/rewrite/spring/SpringServiceLocator.java
+ *
+ * @author Marcelo Fernandes
+ */
+public class SpringBootServiceLocator implements ServiceLocator {
```

Comment:
Does `org.ocpsoft.rewrite.spring.SpringServiceLocator` not work?

Reply:
This class uses `ContextLoader.getCurrentWebApplicationContext()` and [it does not work with spring boot apps](https://stackoverflow.com/questions/22167912/contextloader-getcurrentwebapplicationcontext-always-returns-null). 

`SpringBootServiceLocator` solution uses a static field. Is not the best option to solve this problem, but it works for the moment. I think we could talk to rewrite guys and learn how to integrate better.